src/compiler/scala/tools/nsc/typechecker/Infer.scala

```diff
@@ -148,7 +148,7 @@ trait Infer extends Checkable {
   }
 
   def skipImplicit(tp: Type) = tp match {
-    case mt: MethodType if mt.isImplicit  => mt.resultType
+    case mt: MethodType if mt.isImplicit  => mt.resultType //
```

Comment:
Can you explain the issue here a little more for me? If I try,
```
def foo[A](a: DummyImplicit)(implicit b: DummyImplicit): Array[A] = null
foo: Seq[_]
```
(ie. your second example but with the first parameter list not implicit) I see the exact same error as with both parameter list implicit,
```
multi-implicits.scala:81: error: polymorphic expression cannot be instantiated to expected type;
 found   : [A](a: DummyImplicit)(implicit b: DummyImplicit)Array[A]
 required: Seq[_]
    foo: Seq[_]
    ^
```
Are you saying the the call to `skipImplicit` should just be removed? And instead we should just have,
```
val resTpInst  = ctorTpInst.finalResultType
```